### ISSUE TITLE
Allow granular styling of checkbox, label, and icon in Checkbox and Radio components

### ISSUE THREAD

**Issue by davidhousedev** [chakra-ui/chakra-ui#370]
Thank you for the incredible work! Chakra UI is awesome. :)

**Is your feature request related to a problem? Please describe.**
Thus far, our implementation of Chakra UI has been seamless, with a high degree of freedom in customizing the appearance of components to our designs. However, the implementation of Checkbox and Radio components exposed far less freedom to customize the look and feel..

**Describe the solution you'd like**
Ideally, I would like to be able to be able to inject arbitrary property overrides into the `<Radio>` and the `<Checkbox>` components that will be applied to the `<ControlBox>`, the `<Box as="span">`, and the `<Icon>` child components in both Chakra components.

One API that I could imagine could be something like:
```jsx
<Checkbox propOverrides={'checkbox': {}, 'label': {}, 'icon': {}} />
```

Alternatively, the overrides could be exposed as top-level properties:
```jsx
<Checkbox checkboxProps={} labelProps={} iconProps={} />
```

**Describe alternatives you've considered**
Presently, the workaround I am implementing is to follow the recommendations outlined in https://chakra-ui.com/controlbox however this involves duplicating much of the existing source code in Chakra UI for only a few aesthetic changes in the sub-components of `<Radio>` and `<Checkbox>`.

It would be very helpful to implement the Chakra Checkbox and Radio components, out-of-the-box. However, we need a high degree of style customization.

**Comment by segunadebayo** [comment]
Hi @davidhousedev, I agree that it's a bit painful to update the styles for the Checkbox and Radio components.

In the upcoming TS release, we're looking to make this a lot easier.

**Comment by EllaSharakanski** [comment]
Is there an update on this? Thanks!

**Comment by jackblackCH** [comment]
I am also interested how to create custom checkbox styles and their children. Label checkbox, and all the pseudo styles of them.

**Comment by euanmillar** [comment]
This is a good hack that if you use styled-components:

https://github.com/chakra-ui/chakra-ui/issues/582

**Comment by tausvels** [comment]
Hi. is there a way to custom style the Checkbox component? I want the <span> tag inside the component to have a custom width. this currently can't be achieved with the Size prop as it accepts the predefined string values which are 'sm' or 'md' or 'lg'

**Comment by JackUait** [comment]
Are there any updates on this topic?🤔

**Comment by nelsieborja** [comment]
Checkbox `iconSize` prop seems broken. Tried setting its value but it is not affecting the size of the check icon.

**Comment by szymon-marczuk-arc** [comment]
Any update on this ?

**Comment by mattstobbs** [comment]
In case anyone still has this problem, we were able to update the stylings by updating our theme to extend the base styles found in https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/checkbox.ts

```js
cons theme = {
  components: {
    Checkbox: {
      baseStyle: {
        icon: {
          color: 'white',
        },
        control: {
          border: '1px',
          borderColor: 'gray.300',
          borderRadius: 'base',
          _disabled: {
            borderColor: 'gray.300',
            bg: 'gray.200',
          },
        },
        label: {
          fontWeight: 'medium',
          color: 'gray.900',
        },
      },
    },
  },
};
```

**Comment by nvmnghia** [comment]
Radio styling using @mattstobbs  method doesn't work for me, I still can't set the color `borderColor` of the radio.

**Comment by jchesterman** [comment]
> In case anyone still has this problem, we were able to update the stylings by updating our theme to extend the base styles found in https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/checkbox.ts
> 
> ```js
> const theme = {
>   components: {
>     Checkbox: {
>       baseStyle: {
>         icon: {
>           color: 'white',
>         },
>         control: {
>           border: '1px',
>           borderColor: 'gray.300',
>           borderRadius: 'base',
>           _disabled: {
>             borderColor: 'gray.300',
>             bg: 'gray.200',
>           },
>         },
>         label: {
>           fontWeight: 'medium',
>           color: 'gray.900',
>         },
>       },
>     },
>   },
> };
> ```

This seems to work, but only for certain style props.. i'm currently trying to force the checkbox width and height to 40px, but only the width prop changes, the height seems to be fixed at 1.25rem for the lg size.
